hazelcast/src/test/java/com/hazelcast/internal/partition/TestPartitionUtils.java

```diff
@@ -53,7 +53,7 @@ private TestPartitionUtils() {
     public static PartitionServiceState getPartitionServiceState(HazelcastInstance instance) {
         try {
             Node node = getNode(instance);
-            InternalPartitionServiceImpl partitionService = (InternalPartitionServiceImpl) node.getPartitionService();
```

Comment:
Downcasting is also a problem when using proxied classes (`ClassCastException: Proxy$43 cannot be cast to InternalPartitionServiceImpl` etc.). We already have an internal interface, so it's easy to get rid of this.

Reply:
Yes. Casting leads to coupling and coupling leads to suffering.